### PR TITLE
Add --output-file embossc flag to allow setting exact generated file name

### PR DESCRIPTION
**Pull request by studgeek**
Allow users to specific the exact output header they want from embossc

### CODE REVIEW

File: embossc [hunk 1]
```diff
@@ -46,6 +46,10 @@ def _parse_args(argv):
                       nargs=1,
                       default=".",
                       help="Prefix to use for the generated output file.")
+  parser.add_argument("--output-file",
+                      nargs=1,
+                      help="""Exact file path to use for the generated output
+                              file. Overrides --output-path is ignored.""")
```
Comment (reventlov) [inline embossc:52]: > Overrides --output-path is ignored.

"Only one of --output-path and --output-file may be used", maybe?  And update the --output-path message and throw an error if both are specified.

**Review by reventlov — CHANGES_REQUESTED** [review]

**Comment by reventlov** [comment]
`--output-file` is consistent with `--output-path` and the other `embossc` flags.

File: embossc (hunk 1 above)
Comment (AaronWebster) [reply embossc:52]: That works for me.  And you are correct, there is inconsistent dash/underscore used here which is beyond the scope of this PR, so disregard that comment.

Comment (studgeek) [reply embossc]: I reworked closer to what Aaron suggested since it seems more straightforward relative to existing behavior and  flags. Now  --output-file help is "File name to be used for the generated output file. Defaults to input_file suffixed by '.h'". PTAL

File: embossc [hunk 2]
```diff
@@ -96,10 +102,15 @@ def main(argv):
...
 
-  output_file = os.path.join(flags.output_path[0], flags.input_file[0]) + ".h"
-  os.makedirs(os.path.dirname(output_file), exist_ok=True)
+  if flags.output_file:
+    output_file = flags.output_file[0]
+  else:
+    output_file = flags.input_file[0] + ".h"
+
+  output_filepath = os.path.join(flags.output_path[0], output_file)
+  os.makedirs(os.path.basename(os.path.dirname(output_filepath)), exist_ok=True)
```
Comment (reventlov) [inline embossc:111]: Why did you add the `basename` call?  That's a breaking behavioral change.

**Review by reventlov — CHANGES_REQUESTED** [review]

File: embossc (hunk 2 above)
Comment (studgeek) [reply embossc]: Ah, that is a merge mistake on my part. Removed.

**Review by reventlov — APPROVED** [review]
